engine/src/main/java/org/pentaho/di/job/entries/getpop/MailConnection.java

```diff
@@ -837,46 +825,55 @@ private void handlePart( String foldername, Part part, Pattern pattern ) throws
     }
   }
 
+  @VisibleForTesting
+  static String findValidTarget( String folderName, String fileName ) throws KettleException {

```

Comment:
Why not use `org.apache.commons.io.FilenameUtils` to get the filename without extension and extension.
The code would be cleaner.

Reply:
Agreed - good call. I've updated the PR with your suggestion as well as improved the performance of the algorithm.